src/prototype_room_my.js

```diff
@@ -432,10 +432,26 @@ Room.prototype.executeRoom = function() {
   this.handleTerminal();
   this.handleNukeAttack();
 
-  if (Game.time % 10 === 0) {
+  if(!this.memory.spawnTimer){
+    this.memory.spawnTimer = 0;
+  }
+  var harvesters = this.find(FIND_MY_CREEPS, {
+    filter: function(object) {
+      if (object.memory.role === 'harvester') {
+        return object.memory.base == room.name;
+      }
+      return false;
+    }
+  });
+  if(harvesters.length < 1){
     this.spawnCheckForCreate();
   }
-
+  if(this.memory.spawnTimer > 100 ||  this.energyAvailable == this.energyCapacityAvailable || (this.controller.level *250 < this.energyCapacityAvailable && Game.Time % 50 == 0 ) || this.memory.attackTimer > 15){
+    this.memory.spawnTimer = 0;
+    this.spawnCheckForCreate();
+  } else {
+    this.memory.spawnTimer = this.memory.spawnTimer + 1;
+  }
```

Comment:
I don't think this is a good idea, the spawn should be constantly spawning.
The queue should make sure that the most important creep will be spawned.
The creeps should make sure that they are spawned in the best size possible.

So if your problem is that, creeps are spawned which could have been bigger, if they are spawned later. The logic for that, should live in the creep config logic.